b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/idpSection/otherComponents/idpCreateModal/externalIdentityProvider.tsx

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentityProvider, IdentityProviderConfigureType, IdentityProviderTemplate } from
+    "@b2bsample/business-admin-app/data-access/data-access-common-models-util";
+import { controllerDecodeCreateIdentityProvider } from
+    "@b2bsample/business-admin-app/data-access/data-access-controller";
+import { FormButtonToolbar, FormField } from "@b2bsample/shared/ui/ui-basic-components";
+import { checkIfJSONisEmpty } from "@b2bsample/shared/util/util-common";
+import { LOADING_DISPLAY_BLOCK, LOADING_DISPLAY_NONE, fieldValidate } from
+    "@b2bsample/shared/util/util-front-end-util";
+import { Session } from "next-auth";
+import { useState } from "react";
+import { Form } from "react-final-form";
+import { Loader, Radio, RadioGroup } from "rsuite";
+import FormSuite from "rsuite/Form";
+import styles from "../../../../../../../styles/Settings.module.css";
+
+interface ExternalIdentityProviderProps {
+    session: Session
+    template: IdentityProviderTemplate
+    onIdpCreate: (response: IdentityProvider) => void,
+    onCancel: () => void,
+}
+
+/**
+* 
+* @param prop - `ExternalIdentityProviderProps`
+* 
+* @returns Form to create external idp
+*/
+export default function ExternalIdentityProvider(prop: ExternalIdentityProviderProps) {
+
+    const { session, template, onIdpCreate, onCancel } = prop;
+
+    const [ loadingDisplay, setLoadingDisplay ] = useState(LOADING_DISPLAY_NONE);
+    const [ configureType, setConfigureType ]
+        = useState<IdentityProviderConfigureType>(IdentityProviderConfigureType.AUTO);
+
+    const validate = (values: Record<string, string>): Record<string, string> => {
+        let errors: Record<string, string> = {};
+
+        errors = fieldValidate("application_name", values.application_name, errors);
+        errors = fieldValidate("client_id", values.client_id, errors);
+        errors = fieldValidate("client_secret", values.client_secret, errors);
+
+        switch (configureType) {
+            case IdentityProviderConfigureType.AUTO:
+                errors = fieldValidate("discovery_url", values.discovery_url, errors);
+
```

Comment:
Is this new line required?

Reply:
It's an eslint rule to add a new line before `break`.

Expected blank line before this statement.eslint[@typescript-eslint/padding-line-between-statements](https://typescript-eslint.io/rules/padding-line-between-statements)